File: Data Structures & Algorithms/longest-increasing-subsequence/submission-1.cpp
class Solution {
public:
    map<pair<int,int>,int>mp;
    int dfs(vector<int>&nums,int i, int lidx){
        if(i==nums.size()) return 0;
        if(mp.find({i,lidx})!=mp.end()) return mp[{i,lidx}];
        int ans = dfs(nums,i+1,lidx);
        if(lidx==-1 or nums[i]>nums[lidx]){
            ans = max(ans, 1+dfs(nums,i+1,i));
        }
        mp[{i,lidx}]=ans;
        return ans;
    }
    int lengthOfLIS(vector<int>& nums) {
        return dfs(nums,0,-1);
    }
};

File: Data Structures & Algorithms/majority-element/submission-1.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        ans, cnt = nums[0], 0
        for x in nums:
            if x==ans:
                cnt += 1
            else :
                cnt -=1
                if cnt ==0:
                    ans = x
                    cnt = 1
        return ans